lib/Sema/CSSimplify.cpp

```diff
@@ -10023,8 +10023,13 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
         else
           newTupleTypes.push_back(smallerElt);
       } else {
-        if (largerElt.getType()->isTypeVariableOrMember())
-          recordPotentialHole(largerElt.getType()->getAs<TypeVariableType>());
+        if (largerElt.getType()->isTypeVariableOrMember()) {
+          Type eltType = largerElt.getType();
+          if (auto depMember = eltType->getAs<DependentMemberType>()) {
+            eltType = depMember->getBase();
```

Comment:
What if the base is another `DependentMemberType`? I think we want to continue digging through to the underlying type variable

Reply:
I think it’s only an immediate one is `isTypeVariableOrMember`.

Reply:
Ah it checks in the base type too... We might want to have a version of `recordPotentialHole` which asserts and uses `Type::visit`

Reply:
> We might want to have a version of recordPotentialHole which asserts and uses Type::visit

Right, I'll adjust that :)

Reply:
I was trying to avoid adding `recordPotentialHole(Type)` but I guess we don't have a choice, this would also mean that it supersedes both existing ones.

Reply:
Fixed :)